addons/controllers/clusterbootstrap_controller.go

```diff
@@ -1179,12 +1239,19 @@ func (r *ClusterBootstrapReconciler) updateValuesFromProvider(cluster *clusterap
 			// There are possibilities that current reconciliation loop fails due to various reasons, and during next reconciliation
 			// loop, it is possible that the provider resource has been created. In this case, we want to run update/patch.
 			if apierrors.IsAlreadyExists(err) {
-				// Setting the resource version is because it's been removed at L984 with 'unstructured.RemoveNestedField(newProvider.Object, "metadata")'
-				// apiserver requires that field to be present for concurrency control.
-				newProvider.SetResourceVersion(provider.GetResourceVersion())
-				createdOrUpdatedProvider, err = r.dynamicClient.Resource(*gvr).Namespace(cluster.Namespace).Update(r.context, newProvider, metav1.UpdateOptions{})
+				r.Log.Info(fmt.Sprintf("provider %s/%s already exist, patching its Labels and OwnerReferences fields", newProvider.GetNamespace(), newProvider.GetName()))
+				// Instantiate an empty unstructured and only set ownerReferences and Labels for patching
+				patchObj := unstructured.Unstructured{}
+				patchObj.SetLabels(newProvider.GetLabels())
+				patchObj.SetOwnerReferences(newProvider.GetOwnerReferences())
+				patchData, err := patchObj.MarshalJSON()
+				if err != nil {
+					log.Error(err, fmt.Sprintf("unable to patch provider %s/%s", newProvider.GetNamespace(), newProvider.GetName()), "gvr", gvr)
+					return nil, err
+				}
+				createdOrUpdatedProvider, err = r.dynamicClient.Resource(*gvr).Namespace(cluster.Namespace).Patch(r.context, newProvider.GetName(), types.MergePatchType, patchData, metav1.PatchOptions{})
```

Comment:
Same comment, is this safe? We should be only patching ownerRefs and labels.

Reply:
yes, we should only patch ownerRefs and labels. That is why I added L1243-L1247. Those few lines are doing the following few things:
- create an empty map
- set only metadata.ownerRefs and metadata.labels field
- Marshal to json bytes. I.e., `{"metadata":{"labels":{"foo":"bar"},"ownerReferences":[{"apiVersion":"Foo","kind":"Bar","name":"Zoo","uid":""}]}}`
- Call dynamic client to patch

An alternative is to use `fmt.Sprintf()` to construct the json string manually. Using `unstructured.Unstructured{}` is just to create the patch data in a more "structured", readable and maintainable way.